components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -2702,6 +2702,35 @@ def createExporter(self):
         """
         return ProxyObjectWrapper(self, 'createExporter')
 
+    ####################
+    # Read-only status #
+
+    def getReadOnlyStatus(self):
+        """
+        Get the read-only status of the server.
+
+        :return:  a map of server subsystems each of whose value indicates if
+                  it is read-only
+        """
+        key_prefix = 'omero.cluster.read_only.runtime.'
+        key_regex = '^' + key_prefix.replace('.', '\.')
+        properties = self.getConfigService().getConfigValues(key_regex)
+        return dict((key[len(key_prefix):], value.lower() == 'true')
+                    for key, value in properties.items())
+
+    def isAnyReadOnly(self, *subsystems):
```

Comment:
I'm not familiar with use-cases here but is this a typical use case:
```
conn.isAnyReadOnly( conn.getReadOnlyStatus().keys() )
```
?
If so, it might be handy to have that be the default behaviour if I do ```conn.isAnyReadOnly()``` ?

Reply:
That feels to me to too greatly confuse the semantics: for instance, sometimes adding an argument makes it more likely to be True, sometimes less likely. As it is it tries to be the least surprising in fitting Python's `any([]) == False`. For the use case that you mention one can simply write,
```python
any(conn.getReadOnlyStatus().values())
```

Reply:
OK, fair enough.